pkg/api/v1/defaults.go

```diff
@@ -112,9 +112,10 @@ func SetDefaults_Firmware(obj *Firmware) {
 func SetDefaults_VirtualMachineInstance(obj *VirtualMachineInstance) {
 	// FIXME we need proper validation and configurable defaulting instead of this
 	if _, exists := obj.Spec.Domain.Resources.Requests[v1.ResourceMemory]; !exists {
-		obj.Spec.Domain.Resources.Requests = v1.ResourceList{
-			v1.ResourceMemory: resource.MustParse("8192Ki"),
+		if obj.Spec.Domain.Resources.Requests == nil {
```

Comment:
yeah, I was surprised by that as well but I found out that:
```golang
type mm map[string]int
func main() {
        var aa mm
	fmt.Println(aa["b"])
	if aa == nil {
	   fmt.Println("nil")
	}
}
```

returns:
```
0
nil
``` 

I discovered that because without this block (lines 115-117), tests that don't set requests explode with `panic` on the assignment at line 118

Reply:
@davidvossel Just for reference from
https://blog.golang.org/go-maps-in-action

*Map types are reference types, like pointers or slices, and so the value of m above is nil; it doesn't point to an initialized map. A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that.*